agent/installer/installer.go

```diff
@@ -88,10 +88,7 @@ func (bd *bundleDownloader) DownloadOrPreview(os, k8s, tag string) error {
 // New returns an installer that downloads bundles for the current OS from OCI repository with
 // address bundleRepo and stores them under downloadPath. Download path is created,
 // if it does not exist.
-func New(bundleRepo, downloadPath string, logger logr.Logger) (*installer, error) {
-	if bundleRepo == "" {
-		return nil, fmt.Errorf("empty bundle repo")
-	}
+func New(downloadPath string, logger logr.Logger) (*installer, error) {
```

Comment:
downloadPath goes with bundleRepo. It's a cache of the bundle repo. It's odd that they are decoupled in different methods.

Reply:
I thought downloadPath is a location on disk to cache the repo. Why would it be coupled to the bundleRepo?

Reply:
Indeed "cache **the** repo" - yes. After this change there is no "the repo"

This change makes the cache shareable between repos. If they happen to have different bundles with the same k8s version, this will simply not work as they will clash and will have false hits. Installer assumes 1 repo, 1 local cache. This change is half-making another model.